src/redux/books/booksSlice.js

```diff
@@ -1,44 +1,82 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// MbLdQuSA291U7W3VkOff
+
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/MbLdQuSA291U7W3VkOff/books';
+
+export const getBooks = createAsyncThunk(
+  'books/getBooks',
+  async () => {
+    const res = await axios(baseUrl);
+    return res.data;
+  },
+);
+
+export const postBook = createAsyncThunk(
+  'books/postBook',
+  async (book, thunkAPI) => {
+    try {
+      await axios.post(baseUrl, book);
+      thunkAPI.dispatch(getBooks());
+      const res = thunkAPI.getState().books;
+      return res;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(
+        error?.data?.message || 'Sorry Error Occured!',
+      );
+    }
+  },
+);
+
+export const deleteBookFromApi = createAsyncThunk(
+  'books/deleteBookFromApi',
+  async (id, thunkAPI) => {
+    try {
+      await axios.delete(`${baseUrl}/${id}`);
+      thunkAPI.dispatch(getBooks());
+      const res = thunkAPI.getState().books;
+      return res.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(
+        error?.data?.message || 'Something went wrong!',
+      );
+    }
+  },
+);
```

Comment:
- [ ] The same happens with your delete function:

1. Refactor it to return the book's id.
```js
export const deleteBookFromApi = createAsyncThunk(
  'books/deleteBookFromApi',
  async (id, thunkAPI) => {
    try {
      await axios.delete(`${baseUrl}/${id}`);
      return id;
    } catch (error) {
      return thunkAPI.rejectWithValue(
        error?.data?.message || 'Something went wrong!',
      );
    }
  },
);
```

2. Modify your reducer to filter the books that have a different id:

```js
...
extraReducers: (builder) => {
...
  builder.addCase(deleteBookFromApi.fulfilled, (state, action) => {
    state.books = state.books.filter((book) => book.id !== action.payload);
  })
...
}
```

Please implement this change! 🤓 🔧